src/logger.h

```diff
@@ -26,21 +25,47 @@ enum Severity {
   CRITICAL = 50
 };
 
-class Format {
+enum Info { DATE, SEVERITY, LINE };
+
+// auto date = []() { return std::to_string(GetDate()); }
+// auto line = [](const Format &f) { return std::to_string(f.line); }
+
+struct InputInfo {
+  const char *file, *fn;
+  int line;
+};
+
+// Format("(%:%:%)", date, line, ip);
+// Format(const char *, f -> string...);
+template <const char *Fmt> class Format {
   std::ostream &stream;
-  const char *fn, *file; int line;
   std::lock_guard<std::mutex> lock;
+  InputInfo info;
```

Comment:
ContextInfo -- contains the contextual information of the program "where" the log was created.